engine/resource/src/resource_archive.cpp

```diff
@@ -643,9 +643,9 @@ namespace dmResourceArchive
     {
         int count = dmEndian::ToNetwork(archive->m_EntryDataCount);
         size_t hash_length = dmEndian::ToNetwork(archive->m_HashLength);
-        const uint8_t* end = hashes + count * hash_length;
+        const uint8_t* end = hashes + count * dmResourceArchive::MAX_HASH;
```

Comment:
The end was calculated wrong.

While this caused a buffer overrun (read), it shouldn't have affected the outcome, as the calculation on line 658 `(insert-hashes)/MAX_HASH` takes the correct stride into account, and `insert` will never point beyond the `(hashes + count * MAX_HASH)`.